### DISCUSSION TITLE
ContentType vs Taxonomy built on this ContentType

### DISCUSSION THREAD

**Discussion by m-shanti** [OrchardCMS/OrchardCore#11583] [Q&A] [answered]
Hi,
I created Content Type "Person" and registered several persons (Content Items).
Then I created Taxonomy "Authors" built on Content Type "Person" and I see that I have to register persons in this taxonomy from the beginning. This taxonomy doesn't include so far added persons. I feel I make big mistake in understanding Orchard...

My goal is to have taxonomy Persons and I want to link to persons contained in this Taxonomy in other articles.
I see that Persons created as "free" Content Items (add new Person) has fine urls e.g. "/smith-will" but Content Items added in Taxonomy Persons has long urls e.g "authors/4g9dmb0x24y72xs6kpf6rqy7dm-smith-will".

My additional question is: how to easily add link to other Content Items when you are writing some content? I mean: I'm writing some article. I'm referring in this article to other articles by links in text. Now I have to find the article I want to refer, get its permalink and come back to original article and continue. It's a bit uncomfortable.

Thanks for answers! Good day!

**Comment by Psichorex** [comment, marked as answer]
Taxonomies are more like categories in a way. It's rather used for categorizing content items. e.g. You have a Person content type that could be an Author, Writer, Singer etc. You could choose when creating the person that which of these terms should apply to it. It can be multiple.
If I understand correctly you want to have default Persons to be linked to other places like a Contact Person which is the same everywhere across the whole site. I want to be sure that I really understand your request so here is my visualization: You have groups of Person (a list) and every group has a mentor for example. You want to link that mentor with every member of their own group. Is it what you are looking for? If yes, creating your own Site setting with these default mentors could be a solution. We use site settings for similar reasons. For example storing a contact email which is the same everywhere. You could also add a ContentPickerField to the Person ContentPart and set that you can only choose Persons inside that PickerField. This way you can always choose a Person (optionally multiple) when creating a Person and they will be kind of attached to each other. I think this is the easier and better solution for you if I understand you correctly.
Feel free to update me about your progress.

**Reply by m-shanti** [reply]
Thanks @Psichorex!

Maybe lets divide my "problems":
1. How to share content items between taxonomies. Or how to make that every item of specific content type is automatically member of selected taxonomy
2. Links problem. Let's put it aside. I would like to focus on the first and not mix :)

So the problem 1. is:
- I have content type Person
- I have several content items of this Type
- I create taxonomy Authors built on content type Person
- I would like to have all persons already created in this taxonomy - but I haven't

> Taxonomies are more like categories in a way. It's rather used for categorizing content items. e.g. You have a Person content type that could be an Author, Writer, Singer etc. You could choose when creating the person that which of these terms should apply to it. It can be multiple.

How can I reuse the same items in different taxonomies?

**Reply by Psichorex** [reply]
I don't want to say it is not possible I think you misunderstood the conception of taxonomies. 
You can make a Person have a TaxonomyField for example Job (Job is a Taxonomy here) and you can give that taxonomy several TaxonomyTerms e.g. Uber Driver,  Cook, Janitor etc.
This way when you create a Person there will be a dropdown field called Job where you can set that Person's Job. This is how we use taxonomies. Taxonomies are attached to Content Items and not Content Items to Taxonomies.

**Reply by m-shanti** [reply]
Yes, I must misunderstand the concept.

I think that taxonomy for Persons can be wrong choice.
I created Content Type "Person" which is "rich": name, description and so on.

Then I created Content Type "Book": title, description and I wanted to connect it with Author - which is one of persons.
I thought that taxonomy could be good in this place but I start to understand that not.

Now I used Content Picker Field as you suggested. The Book has content picker to Person. And it's what I searched :)
Thank you! The first my problem is resolved.

Can we come back to the second problem? :
I'm writing some article and I want to place in it links to specific persons, books. How can I do it in some comfort way? Now I'm jumping to the person or book, getting their permalink and paste it as link (e.g. /will-smith). It's a bit uncomfortable. Maybe there is such content-link?
